src/api/check.ts

```diff
@@ -1,17 +1,18 @@
+import chalk from 'chalk'
 import { CheckOptions, RawDependency, PackageMeta, DependencyFilter, RangeMode, DependencyResolvedCallback } from '../types'
 import { loadPackages, writePackage } from '../io/packages'
 import { resolvePackage } from '../io/resolves'
-
+import { TableLogger } from '../log'
 export interface CheckEventCallbacks {
   afterPackagesLoaded?: (pkgs: PackageMeta[]) => void
   beforePackageStart?: (pkg: PackageMeta) => void
-  afterPackageEnd?: (pkg: PackageMeta) => void
+  afterPackageEnd?: (pkg: PackageMeta) => boolean
   beforePackageWrite?: (pkg: PackageMeta) => boolean | Promise<boolean>
   afterPackageWrite?: (pkg: PackageMeta) => void
   onDependencyResolved?: DependencyResolvedCallback
 }
 
-export async function CheckPackages(options: CheckOptions, callbacks: CheckEventCallbacks = {}) {
+export async function CheckPackages(options: CheckOptions, tableLogger: TableLogger, callbacks: CheckEventCallbacks = {}) {
```

Comment:
Tend to make this function reusable for different env, should be cli-free. Maybe it's better to move it outside and utilize the callbacks to update the counter?

Reply:
Can you revert the argument for the logger?